### PR TITLE
Feature/custom profile

### PR DESCRIPTION
**Pull request by hiddestokvis**
Added an extra option to merge custom fields from the Social Media platform to the user model.

In provider.json add an option:

```
"profileToUserFields": {
   "first_name": "firstName",
   keyInUserModel : keyInSocialMediaProfile
}
```

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by jannyHou** [comment]
@hiddestokvis Thank you for contribution! could you rebase your code and provide unit test to verify your feature? Thanks!

I will run ci test for you after rebase.

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by hiddestokvis** [comment]
@jannyHou the branch is rebased and I have added unit testing + made the necessary changes to pass all tests

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@raymondfeng This feature is ready for review, all test cases pass.
brief feature description: https://github.com/strongloop/loopback-component-passport/pull/91#issue-108672611

Please review it, thank you.

**Comment by jannyHou** [comment]
@raymondfeng ^^^ Please review it when you have time :) Thanks.

File: lib/models/user-identity.js [hunk 1]
```diff
@@ -55,6 +55,32 @@
        username: username,
        password: password,
      };
+
+     if (typeof options.profileToUserFields !== 'undefined') {
+       for (var key in options.profileToUserFields) {
+         var fieldName = options.profileToUserFields[key];
+
+         if (fieldName.search('.') !== -1) {
```
Comment (raymondfeng) [inline lib/models/user-identity.js]: Do you want to test if fieldName contains `.`? If so, your code is NOT working as `.` will match any char. Use one of the following:
- `fieldName.indexOf('.') !== -1`
- `fieldName.search(/\./) !== -1`
- `fieldName.search('\\.') !== -1`

File: test/model.user-identity.test.js [hunk 2]
```diff
@@ -23,13 +23,21 @@ describe('UserIdentity', function() {
...
-      ], id: 'f123', username: 'xyz',
+      ], id: 'f123', username: 'xyz', firstName: 'abc', lastName: '123',
       }, { accessToken: 'at1', refreshToken: 'rt1' },
-      { autoLogin: false },
+      {
+        autoLogin: false,
+        profileToUserFields: {
+          firstName: 'firstName',
+          lastName: 'lastName',
+        },
```
Comment (raymondfeng) [inline test/model.user-identity.test.js]: Can we add a case to cover `a.b` convention?

**Comment by jannyHou** [comment]
Any updates @hiddestokvis ? And could you apply feedback from Raymond and rebase your code? Thanks.

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**